src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -60,19 +64,39 @@
 
     private String authMethod;
 
+    private static final String secPropName =
+        "jdk.httpdigest.defaultDisabledAlgorithms";
+
     private static final String compatPropName = "http.auth.digest." +
         "quoteParameters";
 
     // default set of disabled message digest algorithms that may not be
     // used for proxy connections, or plain text http server connections
 
-    // This will probably be expanded to include SHA-1 eventually
-    private static final Set<String> defDisabledAlgs =
-        Set.of("MD5");
+    private static final Set<String> defDisabledAlgs = getDefaultAlgs();
+
+    private static Set<String> getDefaultAlgs() {
+        Set<String> disabledAlgs = new HashSet<>();
+        @SuppressWarnings("removal")
+        String secprops = AccessController.doPrivileged(
+            new PrivilegedAction<>() {
+                public String run() {
+                    return Security.getProperty(secPropName)
+                                   .replaceAll("\\s", "")
+                                   .toUpperCase();
+                }
+            }
+        );
+        String[] algorithms = secprops.split(",");
+        for (String algorithm : algorithms) {
+            disabledAlgs.add(algorithm);
+        }
+        return Collections.unmodifiableSet(disabledAlgs);
```

Comment:
Could use `Set.copyOf()` ? Or is there anything that might call contains(null)? Also should strings be converted to upper case like below?